src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -325,3 +333,76 @@ func (c controller) sendCompleteInstallation(isSuccess bool, errorInfo string) {
 	}
 	c.log.Infof("Done complete installation step")
 }
+
+// get pod logs,
+// write tar.gz to pipe in a routine
+// upload tar.gz from pipe to assisted service.
+// close read and write pipes
+func (c controller) uploadPodLogs(podName string, namespace string) error {
+	c.log.Infof("Uploading logs for %s in %s", podName, namespace)
+	podLogs, err := c.kc.GetPodLogs(namespace, podName, logsSinceInSec)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get logs of pod %s", podName)
+	}
+	if podLogs == "" {
+		return errors.Errorf("no logs of %s were found", podName)
+	}
+	pr, pw := io.Pipe()
+	defer pr.Close()
+
+	go func() {
+		defer pw.Close()
+		err = utils.AddStringToTarGz(pw, podLogs, fmt.Sprintf("%s.logs", podName))
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to upload")
+		}
+	}()
+	// if error will occur in goroutine above
+	//it will close writer and upload will fail
+	err = c.ic.UploadLogs(c.ClusterID, models.LogsTypeController, pr)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to upload logs")
+	}
+	return nil
+}
+
+// Uploading logs every 5 minutes
+// We will take logs of assisted controller and upload them to assisted-service
+// by creating tar gz of them.
+func (c *controller) UploadControllerLogs(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	c.log.Infof("Start sending logs")
+	podName := ""
+	ticker := time.NewTicker(LogsTimeout)
+	for {
+		select {
+		case <-ctx.Done():
+			if podName != "" {
+				c.log.Infof("Upload logs before exit")
+				_ = c.uploadPodLogs(podName, c.Namespace)
+
+			}
+			c.log.Infof("Done uploading logs")
+			return
+		case <-ticker.C:
+			if podName == "" {
+				pods, err := c.kc.GetPods(c.Namespace, map[string]string{"job-name": "assisted-installer-controller"})
+				if err != nil {
+					c.log.WithError(err).Warnf("Failed to get controller pod name")
+					continue
+				}
+				if len(pods) < 1 {
+					c.log.Infof("Didn't find myself, something strange had happened")
```

Comment:
Is that possible? Can it be changed when running in a loop? Shouldn't be outside of the loop?

Reply:
it only means that for some strange reason there is no pod.
just didn't want to fail here on panic while trying to get pod[0]